mono/metadata/locales.c

```diff
@@ -308,9 +308,9 @@ construct_culture (MonoCultureInfoHandle this_obj, const CultureInfoEntry *ci, M
 	mono_handle_setval (this_obj, lcid, ci->lcid);
 
 #define SET_STR_FIELD(obj,field,domain,expr,error) do {					\
-	MonoStringHandle _str = mono_string_new_handle ((domain), (expr), (error));	\
+	MonoString *_str = mono_string_new_checked ((domain), (expr), (error));	\
 	return_val_if_nok (error, FALSE);										\
-	MONO_HANDLE_SET ((obj), field, _str);	\
+	MONO_HANDLE_SETRAW ((obj), field, _str);	\
```

Comment:
I guess that is a way to do it.

Reply:
I believe there is a goal to cut `mono_string_new_checked` use to zero, not increase it.
There are 46 of them.
But it can wait. This will trip the static analysis, but is ok.
Perhaps time to commit my `assign` pattern?
i.e. `mono_string_new_assign` that takes a handle, that can be reused.
Note that the guidance to use `MONO_HANDLE_ASSIGN` feels inadequate too.
Anyway I guess @lambdageek will weigh in.
My vote again is for `_assign`.

Reply:
Eh, `MONO_HANDLE_ASSIGN` is handle to handle actually so I guess ok.
@lambdageek I think therefore will want function enter/exit here.
You still create 680 handles, but the peak will be reasonably low.
I still like _assign as that cuts it down 1.